app/types.py

```diff
@@ -1,11 +1,42 @@
+from dataclasses import asdict, dataclass
 from datetime import datetime
-from typing import Optional
+from typing import List, NewType, Optional
 
-from app.encryption import NotificationDictToSign
 from app.models import Job, NotificationType, Service
+from app.queue import QueueMessage
 
+SignedNotification = NewType("SignedNotification", str)
+SignedNotifications = NewType("SignedNotifications", List[SignedNotification])
 
-class VerifiedNotification(NotificationDictToSign):
+
+@dataclass
+class PendingNotification(QueueMessage):
+    # todo: remove duplicate keys
+    # todo: remove all NotRequired and decide if key should be there or not
+    id: str
+    template: str  # actually template_id
+    service_id: str
+    template_version: int
+    to: str  # recipient
+    personalisation: Optional[dict]
+    simulated: bool
+    api_key: str
+    key_type: str  # should be ApiKeyType but can't import that here
+    client_reference: Optional[str]
+    reply_to_text: Optional[str]
+
+    def to_dict(self) -> dict:
+        """Convert the object to a dictionary."""
+        return asdict(self)
+
+    @classmethod
+    def from_dict(cls, data: dict):
+        """Create an object from a dictionary."""
+        return cls(**data)
```

Comment:
Looking at the doc for dataclasses: https://docs.python.org/3/library/dataclasses.html

It looks like that decorater will add an `__init__` method where you can pass in all the data you have defined above. So you could create an instance like this:
```
pending_notification = PendingNotification(id=x.id, template=x.template, etc )
```
instead of:
```
pending_notification = PendingNotification.from_dict({"id": x.id, "template": x.template, etc })
```
if you want.